File: topics/arrays/exercises/InsertAt.pas
procedure ????(var data: array of Integer; param3, param4: Integer);
var
    i: Integer;
begin    
    for i := High(data) downto param3 + 1 do
    begin
        data[i] := data[i - 1];
    end;
    
    data[param3] := param4;
end;

File: topics/arrays/exercises/insert-at.c
void ????(int data[], int sz, int param3, int param4)
{
    int i;
    
    for (i = sz - 1; i > param3; i--)
    {
        data[i] = data[i - 1];
    }
    
    data[param3] = param4;
}